runtime/lua/vim/filetype.lua

```diff
@@ -1571,11 +1736,26 @@ local pattern = {
   ['.*/%.config/git/config'] = 'gitconfig',
   ['.*%.git/config%.worktree'] = 'gitconfig',
   ['.*%.git/worktrees/.*/config%.worktree'] = 'gitconfig',
+  ['/git/config'] = function(path, bufnr)
```

Comment:
If the pattern contains a `/` character then the pattern is matched against the full file path, so this will never match. Needs to be:

```suggestion
  ['.*/git/config'] = function(path, bufnr)
```